README.md

```diff
@@ -3,6 +3,13 @@
 ## Description
 [Grafana Agent](https://github.com/grafana/agent) is a telemetry collector for sending metrics, logs, and trace data to the opinionated Grafana observability stack.
 
+As a single entry point to the observability stack, the grafana agent charm brings several conveniences when deployed inside a monitored cluster:
+- Charms are related to the grafana agent charm, instead of to prometheus and loki individually.
```

Comment:
I'm still somewhat skeptical about this as a benefit.

In some cases yes, in some cases no, but I suppose that, personally, I see three major use cases:

1. Adding it as a subordinate on a VM/machine instance where there are multiple services/logfiles of interest. Being able to have the agent read a bunch of logs from disparate services as well as scrapes and trace is a big plus. Grafana Agent on a physical/virtual server can collect everything all-at-once. For "single purpose" things like containers, this is arguably less valuable.
2. Having it sit on the "edge" similar to cos-proxy in order to scrape things on the "other side" (a different network interface/VLAN/whatever) and do the same.
3. As an intermediary step for self-monitoring which can more or less sidestep the problems inherent in both remote_write and remote_write (not that they cannot be in the agent also), which are primarily around relabeling and alerts, especially that a backoff on a remote read will flag _all_ metrics as failed even if half pull successfully, and as a "canary in the coalmine" as a secondary monitoring source.

It may be convenient to use grafana agent instead of prometheus and loki individually, but it comes with several risks. 

* A monolithic configuration where a global config change for metrics may clobber others, or a bad config for something may cause the entire thing to fail to load. 
* A loss in precision when something goes wrong. Imagine that the application is accidentally removed (or a relation removed). 1000 alerts fire at once. Was there a network split/segmentation? Was the agent down? It the disk on something full? You have no way to check, because all your eggs are in one basket.
* A loss in both paralllelism for pushing logs to loki/scraping endpoints from Prometheus, and a risk that a failed ingress will instantly stop all observability, whereas a failed ingress would, mostly, have no effect on Prometheus (operating in the other direction).

From my POV, other than the 3 cases on top, my order of preference for resiliency would be:

* Relate to grafana agent, prometheus, _and_ Loki.
* Relate to Prometheus and either Grafana agent or Loki
* Grafana agent alone.

Reply:
Added deployment scenario (rendered [here](https://github.com/canonical/grafana-agent-k8s-operator/blob/feature/doc_update/README.md)). Is that what we all have in mind?